fast64_internal/oot/oot_utility.py

```diff
@@ -602,3 +633,279 @@ def oot_utility_register():
+	if tiedParam is None or actorType is None:
+		return True
+	else:
+		tiedList = tiedParam.split(',')
+		if actorType is not None:
+			return actorType in tiedList
+	return False
+
+def getActorString(elem, detailedProp, field, lenProp, lenSwitch, lenBool, lenEnum, paramTarget):
+	'''Returns the current actor's parameter string'''
+	shiftTmp = 0
+	params = []
+	paramPart = ""
+	strMask = elem.get('Mask')
+	target = elem.get('Target')
+	if target is None: target = "Params"
+	if elem.tag != 'Parameter' and strMask is not None:
+		mask = int(strMask, base=16)
+		shiftTmp = len(f'{mask:016b}') - len(f'{mask:016b}'.rstrip('0'))
+		shift = f'{shiftTmp}'
+		if elem.tag == 'Flag':
+			elemType = elem.get('Type')
+			if elemType == 'Chest' and target == paramTarget:
+				paramPart = getAndFormatActorParameterPart(detailedProp, field + '.chestFlag', shift, strMask)
+				if paramPart is not None:
+					params.append(paramPart)
+			elif elemType == 'Collectible' and target == paramTarget:
+				paramPart = getAndFormatActorParameterPart(detailedProp, field + '.collectibleFlag', shift, strMask)
+				if paramPart is not None:
+					params.append(paramPart)
+			elif elemType == 'Switch' and target == paramTarget:
+				i = int(elem.get('Index'), base=10)
+				if (i >= 1 and i <= int(lenSwitch, base=10)) and (target == paramTarget):
+					paramPart = getAndFormatActorParameterPart(detailedProp, field + f'.switchFlag{i}', shift, strMask)
+					if paramPart is not None:
+						params.append(paramPart)
+		elif elem.tag == 'Property' and elem.get('Name') != 'None':
+			i = int(elem.get('Index'), base=10)
+			if (i >= 1 and i <= int(lenProp, base=10)) and (target == paramTarget):
+					paramPart = getAndFormatActorParameterPart(detailedProp, (field + f'.props{i}'), shift, strMask)
+					if paramPart is not None:
+						params.append(paramPart)
+		elif elem.tag == 'ChestContent' and target == paramTarget:
+			if shift != '0':
+				params.append(f"(({detailedProp.itemChest} << {shift}) & 0x{mask:X})")
+			else:
+				params.append(f"({detailedProp.itemChest} & 0x{mask:X})")
+		elif elem.tag == 'Message' and target == paramTarget:
+			if shift != '0':
+				params.append(f"(({detailedProp.naviMsgID} << {shift}) & 0x{mask:X})")
+			else:
+				params.append(f"({detailedProp.naviMsgID} & 0x{mask:X})")
+		elif elem.tag == 'Collectible' and target == paramTarget:
+			paramPart = getAndFormatActorParameterPart(detailedProp, field + '.collectibleDrop', shift, strMask)
+			if paramPart is not None:
+				params.append(paramPart)
+		elif elem.tag == 'Bool':
+			i = int(elem.get('Index'), base=10)
+			if (i >= 1 and i <= int(lenBool, base=10)) and (target == paramTarget):
+					paramPart = getAndFormatActorParameterPart(detailedProp, (field + f'.bool{i}'), shift, strMask)
+					if paramPart is not None:
+						params.append(paramPart)
+		elif elem.tag == 'Enum':
+			i = int(elem.get('Index'), base=10)
+			if (i >= 1 and i <= int(lenEnum, base=10)) and (target == paramTarget):
```

Comment:
isn't `i >= 1 and i <= int(lenEnum, base=10)` always supposed to be true? or at least if it as false it would be an issue

Reply:
yeah you're right, it's always true since it's in the actorNode/elem loop